pillarbox-demo-shared/src/main/java/ch/srgssr/pillarbox/demo/shared/ui/player/metrics/StatsForNerdsViewModel.kt

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.demo.shared.ui.player.metrics
+
+import android.app.Application
+import androidx.annotation.StringRes
+import androidx.lifecycle.AndroidViewModel
+import androidx.media3.common.VideoSize
+import ch.srgssr.pillarbox.demo.shared.R
+import ch.srgssr.pillarbox.player.analytics.metrics.PlaybackMetrics
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import java.text.NumberFormat
+import kotlin.collections.sum
+import kotlin.time.Duration
+
+/**
+ * [ViewModel][androidx.lifecycle.ViewModel] for the "Stats for Nerds" screen.
+ */
+class StatsForNerdsViewModel(application: Application) : AndroidViewModel(application) {
+    private val _indicatedBitRates = MutableStateFlow(BitRates.Empty)
+
+    /**
+     * Provides information about the indicated bit rates.
+     */
+    val indicatedBitRates: StateFlow<BitRates> = _indicatedBitRates
+
+    private val _information = MutableStateFlow(emptyMap<String, String>())
+
+    /**
+     * Provides information about the current session.
+     */
+    val information: StateFlow<Map<String, String>> = _information
+
+    private val _observedBitRates = MutableStateFlow(BitRates.Empty)
+
+    /**
+     * Provides information about the observed bit rates.
+     */
+    val observedBitRates: StateFlow<BitRates> = _observedBitRates
+
+    private val _stalls = MutableStateFlow(Stalls.Empty)
+
+    /**
+     * Provides information about stalls.
+     */
+    val stalls: StateFlow<Stalls> = _stalls
+
+    private val _startupTimes = MutableStateFlow(emptyMap<String, String>())
+
+    /**
+     * Provides information about the startup times.
+     */
+    val startupTimes: StateFlow<Map<String, String>> = _startupTimes
+
+    private val _volumes = MutableStateFlow(DataVolumes.Empty)
+
+    /**
+     * Provides information about volumes.
+     */
+    val volumes: StateFlow<DataVolumes> = _volumes
+
+    /**
+     * The latest playback metrics.
+     */
+    var playbackMetrics: PlaybackMetrics? = null
+        set(value) {
+            if (field == value) {
+                return
+            }
+
+            field = value
+
+            if (value == null) {
+                return
+            }
+
+            _indicatedBitRates.update {
+                BitRates(
+                    data = it.data + (value.indicatedBitrate / TO_MEGA),
```

Comment:
Data incremented endlessly if we show chart with live during 10h. We maybe should trim to CHART_MAX_POINTS ?